### ISSUE TITLE
Question about std.size and label.size

### ISSUE THREAD

**Issue by Mordoal** [ZhouCX117/UAED#17]
I'm training UAED with my own datasets on png format, however i found that the std.size and label.size are different when calculating the cross_entropy_loss_RCF where std.size is [4, 1, 384, 544], and label.size is [4, 1, 3, 384, 544]. I noticed the model output is set to be [batchsize, classes, h, w]. Can you help me figure out the reason that the size couldn't match? Am I right in the model output size? 
best wishes, thx

**Comment by Mordoal** [comment]
> It seems that you read the labels wrongly. I unsqueeze the label in line 65 of 'data_loader_one_random.py' to change the shape from [H, W] to [1, H, W]. So you should ensure your label shape is [H, W] rather than [3, H, W].

I cvt the image to [H,W]. but after i transform the dataset by code below
self.transform = transforms.Compose([transforms.ToTensor(), 
                                             transforms.Resize([384, 544])])
and do label = label.float()
                label = label.unsqueeze(0)
the label.size become [4, 1, 1, 384, 544], still didn't match.

**Comment by Mordoal** [comment]
> > It seems that you read the labels wrongly. I unsqueeze the label in line 65 of 'data_loader_one_random.py' to change the shape from [H, W] to [1, H, W]. So you should ensure your label shape is [H, W] rather than [3, H, W].
> 
> I cvt the image to [H,W]. but after i transform the dataset by code below self.transform = transforms.Compose([transforms.ToTensor(), transforms.Resize([384, 544])]) and do label = label.float() label = label.unsqueeze(0) the label.size become [4, 1, 1, 384, 544], still didn't match.

i refer to [RCF_Pytorch_Updated](https://github.com/balajiselvaraj1601/RCF_Pytorch_Updated/tree/master)
/data_loader.py and add the transformation, it seems to work!


I think i run it with no bug once since "Epoch: [0/20][0/111] Time 1.907 (avg:1.907) Loss nan (avg:nan) 
0.0 nan" has showed in console.
but one other error occured when it comes to "outputs_dist=Independent(Normal(loc=mean, scale=std+0.001), 1)" 
Expected parameter loc (Tensor of shape (4, 1, 384, 544)) of distribution Normal(loc: torch.Size([4, 1, 384, 544]), scale: torch.Size([4, 1, 384, 544])) to satisfy the constraint Real(), but found invalid values:
they are full of nan.
can u help with this? many thx!
